roachtest: misc fixes

tbg opened the pull request:
Fixes https://github.com/cockroachdb/cockroach/issues/94272.
Fixes https://github.com/cockroachdb/cockroach/issues/94266.

Epic: none
Release note: None

tbg committed:
roachtest: fix follower-reads

Epic: none
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/94361)
<!-- Reviewable:end -->

tbg committed:
roachtest: add missing paren in network

Fixes https://github.com/cockroachdb/cockroach/issues/94204.

Epic: none
Release note: None

tbg committed:
roachtest: fix error depth

Errors were reported like this:

> (test_impl.go:291).Fatal: pq: invalid syntax @1

but that line is the impl of `Fatal`. We want to walk up one more frame.

Epic: none
Release note: None

srosenberg reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/94361#-:-NKQiznBCQU_HyjA2dTA:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @herkolategan and @renatolabs)



<!-- Sent from Reviewable.io -->

healthy-pod commented:
Re-borsing after `license/cla` re-delivery.

bors r=erikgrinaker
